code_samples/api/rest_api/create_image.json.php

```diff
@@ -65,101 +70,75 @@
     ],
 ];
 
-$curl = curl_init();
-$responseHeaders = [];
-
-curl_setopt_array($curl, [
-    CURLOPT_USERPWD => "$username:$password",
-    CURLOPT_URL => "$baseUrl/content/objects",
-    CURLOPT_POST => true,
-    CURLOPT_POSTFIELDS => json_encode($data),
-    CURLOPT_HTTPHEADER => [
-        'Content-Type: application/vnd.ibexa.api.ContentCreate+json',
-        'Accept: application/vnd.ibexa.api.ContentInfo+json',
-    ],
-    CURLOPT_RETURNTRANSFER => true,
-    CURLOPT_HEADERFUNCTION => static function ($curl, $header) {
-        global $responseHeaders;
-        $responseHeaders[] = $header;
-
-        return strlen($header);
-    },
+$client = HttpClient::createForBaseUri($baseUrl, [
+    'auth_basic' => [$username, $password],
 ]);
 
-$response = curl_exec($curl);
-
-if ($error = curl_error($curl)) {
-    echo "cURL error: $error\n";
-    curl_close($curl);
+try {
+    $response = $client->request('POST', "$baseUrl/content/objects", [
```

Comment:
You can skip `$baseUrl` part because it'll be automatically applied by HttpClient since it was initialized with base URL already.

Reply:
@webhdx 
I tried to remove `$baseUrl` with or without the leading slash, I always got 404.
The description of `createForBaseUri` is "Creates a client that adds options (e.g. authentication headers) only when the request URL matches the provided base URI." which implies (from my PoV) that this client can access anything and doesn't add the $baseUrl, it just adds the `auth_basic` header if base URIs of request and creation match.